plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergModule.java

```diff
@@ -177,4 +177,17 @@ public ExecutorService createScanPlanningExecutor(CatalogName catalogName, Icebe
                 config.getSplitManagerThreads(),
                 daemonThreadsNamed("iceberg-split-manager-" + catalogName + "-%s"));
     }
+
+    @Provides
+    @Singleton
+    @ForIcebergSystemTableScanPlanning
+    public ExecutorService createSystemTableScanPlanningExecutor(CatalogName catalogName, IcebergConfig config)
```

Comment:
I don't see a strong reason why this should be a separate thread pool from `createScanPlanningExecutor`.
System tables read iceberg metadata in the same way as normal table scans. So why would we run them in a separate thread pool ?

Reply:
Unfortunately, deploying a new version to the affected environment is a little challenging. 

In the meantime, I can:

1. Update the PR to use the same thread pool (because I think all iceberg scan planning should use a thread pool managed by Trino where possible).
2. See if I can generate some tests/metrics that would support introducing another thread pool.

Thoughts?

Reply:
we can do that

Reply:
Number 1 is done. Feel free to merge since I'm not sure when I'll get to number 2.